doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,247 @@
+{
+-   public partial class ParameterClient
++   public partial class Parameter
+    {
+-       protected ParameterClient()
++       protected Parameter()
+-       public ParameterClient(AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
++       internal Parameter(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, AzureKeyCredential keyCredential, Uri endpoint) {}
+    }
+}
+
+```
+
+</details>
+
+### Client hierarchy
+
+Each client can be made a subclient of another client (without circular dependency) using CodeGenClientAttribute.ParentClient parameter.
+Since CodeGenClientAttribute is applied to the client type, user will explicitly specify new client type name. According to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients), type should have a name without "Client" suffix and an internal constructor. A factory method will be added to the type, specified in ParentClient.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\RootClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class RootClient
+    {
+        public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
+    }
+}
+
+//Generated\ParameterClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class ParameterClient
+    {
+        public ParameterClient(AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
+    }
+}
+```
+
+**Add client hierarchy customization (Customizations.cs):**
+
+```C#
+
+//Customizations.cs
+using Azure.Core;
+
+namespace Azure.Service.SubClients
+{
+    [CodeGenClient("ParameterClient", ParentClient = typeof(RootClient))]
+    public partial class Parameter { }
+}
+```
+
+**Generated code after:**
+
+```diff
+namespace Azure.Service.SubClients
+{
+-   public partial class SubClientsClientOptions : ClientOptions{}
++   public partial class RootClientOptions : ClientOptions{}
+}
+//Parent client: RootClient Generated\RootClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class RootClient
+    {
+-        public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null)
++        public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, RootClientOptions options = null)
+         {
+             Argument.AssertNotNullOrEmpty(cachedParameter, nameof(cachedParameter));
+             Argument.AssertNotNull(credential, nameof(credential));
+             endpoint ??= new Uri("http://localhost:3000");
+-            options ??= new SubClientsClientOptions();
++            options ??= new RootClientOptions();
+
+             ClientDiagnostics = new ClientDiagnostics(options);
+             _keyCredential = credential;
+             
+         }
+
++        private Parameter _cachedParameter0;
```

Comment:
It's kind of confusing that this appears in the diff.  My preference would be to show just the API differences and not the implementation details.

Reply:
removed